dvc/commands/experiments/ls.py

```diff
@@ -11,19 +12,29 @@
 class CmdExperimentsList(CmdBase):
     def run(self):
         name_only = self.args.name_only
+        sha_only = self.args.sha_only
+        git_remote = self.args.git_remote
+        if sha_only and git_remote:
+            raise InvalidArgumentError("--sha-only not supported with git_remote.")
         exps = self.repo.experiments.ls(
             all_commits=self.args.all_commits,
             rev=self.args.rev,
             num=self.args.num,
-            git_remote=self.args.git_remote,
+            git_remote=git_remote,
         )
 
         for baseline in exps:
-            if not name_only:
+            if not (name_only or sha_only):
                 ui.write(f"{baseline}:")
-            for exp_name in exps[baseline]:
-                indent = "" if name_only else "\t"
-                ui.write(f"{indent}{exp_name}")
+            for exp_name, rev in exps[baseline]:
+                if name_only:
+                    ui.write(exp_name)
+                elif sha_only:
+                    ui.write(rev)
+                elif rev:
+                    ui.write(f"\t{rev} [{exp_name}]")
```

Comment:
```suggestion
                    ui.write(f"\t{rev[:7]} [{exp_name}]")
```

(see my other comment about keeping the entire rev in the internal API call)